### ISSUE TITLE
Add support for cloning git repositories using SSH

### ISSUE THREAD

**Issue by Angelmmiguel** [vmware-labs/wasm-workers-server#151]
### Is your feature request related to a problem? Please describe.

After closing #147, you can use a remote git repository as a valid location. However, it only clone repos over HTTP/S. The following command doesn't work:

```
wws git@github.com:vmware-labs/wasm-workers-server.git
```

### Describe the solution you'd like

I want to clone git repositories using the SSH protocol too. It will require to detect these kind of URIs, load the SSH key and try to pull it. For the SSH key, I need to evaluate the different options. From the `git2-rs` library, you can load it [from a file, the memory or a ssh-agent](https://docs.rs/git2/latest/git2/struct.Cred.html).

You may need to add more `git` options to the CLI so you can get SSH keys location, agent usernames, etc.

**Comment by gedw99** [comment]
This ssh protocol would be nice addition.

madding the ability to flag the auth provider too would be a nice to have. For example ssh keys.

Then git config —local  xxxx to tell it the path to your ssh key.

**Comment by Angelmmiguel** [comment]
Yes! That's the main reason why we created a separate task for this. We would need to add more configuration options for managing auth methods. I will try to complete the implementation details on the issue to discuss about them 👍
